kaldb/src/main/java/com/slack/kaldb/chunk/ChunkManager.java

```diff
@@ -296,7 +299,39 @@ public void rollOverActiveChunk() {
                         query.startTimeEpochMs / 1000, query.endTimeEpochMs / 1000))
             .map(
                 (chunk) ->
-                    CompletableFuture.supplyAsync(() -> chunk.query(query), queryExecutorService)
+                    CompletableFuture.supplyAsync(
+                            () -> {
+                              ScopedSpan span =
+                                  Tracing.currentTracer()
+                                      .startScopedSpan("ReadWriteChunkImpl.query");
+                              span.tag("chunkId", chunk.info().chunkId);
+                              span.tag(
+                                  "chunkCreationTimeSecsSinceEpoch",
+                                  String.valueOf(chunk.info().getChunkCreationTimeEpochSecs()));
+                              span.tag(
+                                  "chunkLastUpdatedTimeSecsEpochSecs",
+                                  String.valueOf(
+                                      chunk.info().getChunkLastUpdatedTimeSecsEpochSecs()));
+                              span.tag(
+                                  "dataStartTimeEpochSecs",
+                                  String.valueOf(chunk.info().getDataStartTimeEpochSecs()));
+                              span.tag(
+                                  "dataEndTimeEpochSecs",
+                                  String.valueOf(chunk.info().getDataEndTimeEpochSecs()));
+                              span.tag(
+                                  "chunkSnapshotTimeEpochSecs",
+                                  String.valueOf(chunk.info().getChunkSnapshotTimeEpochSecs()));
+                              span.tag("numDocs", String.valueOf(chunk.info().getNumDocs()));
+                              span.tag("chunkSize", String.valueOf(chunk.info().getChunkSize()));
+                              span.tag("readOnly", String.valueOf(chunk.isReadOnly()));
+                              
+                              try {
+                                return chunk.query(query);
+                              } finally {
+                                span.finish();
+                              }
+                            },
+                            RequestContext.makeContextPropagating(queryExecutorService))
```

Comment:
I can't comment on the snippet below ( since nothing has changed ) but can we add a span in the exceptionally block ?